store/src/state.rs

```diff
@@ -320,13 +321,17 @@ impl State {
         Ok(())
     }
 
+    /// Queries a [BlockHeader] from the database.
+    ///
+    /// If [None] is given as the value of `block_num`, the latest [BlockHeader] is returned.
     pub async fn get_block_header(
         &self,
         block_num: Option<BlockNumber>,
     ) -> Result<Option<block_header::BlockHeader>, anyhow::Error> {
         self.db.select_block_header_by_block_num(block_num).await
     }
 
+    /// Generates membership proofs for each one of the `nullifiers`.
```

Comment:
Should we add a note to the effect of: "The membership proofs are generated against the most recent state of the nullifier tree and will become invalid as soon as the tree is updated".

Also note for the future: I wonder if as a part of this response we should also return the block number for the block against which the proofs were generated. Otherwise, the client may not be able to tell for which root the proofs are valid.

Reply:
changed